### DISCUSSION TITLE
Shelly 1PM vs Vue2 energy resets

### DISCUSSION THREAD

**Discussion by copekyle** [emporia-vue-local/esphome#112] [Q&A]
I have a vue monitoring circuits in my home and then shelly 1pm's measuring large load items. I'm bringing all the values into home assistant. I'm dealing with one circuit weirdness right now. My fridge circuit also has my outdoor outlets on it and so I'm measuring the fridge with a 1PM and then getting real time numbers by subtracting the power of the fridge from the total circuit value. This works for the most part. I have some error checking because the measurements aren't always in sync and if the fridge is larger than the whole circuit I discard the value.

Now the problem, I'd like to also do this with the Daily energy values. Both my devices report the same time. When I subtract the fridge energy from the total energy, I get a negative 1800 Wh. Note, the 1PM gives this value in kWh and the vue is Wh. I have compensated for that. I can't find a reason how these would be so different. Any ideas?

**Comment by copekyle** [comment]
I've spent 3 hours on this and found a simple answer. :( The Shelly devices don't reset unless power cycled. I recently had a power outage and so the energy numbers looked realistic for the device. Looks like I'll be calculating my own daily usage numbers for the shelly devices.

**Reply by nekorevend** [reply]
ESPHome supports Shelly devices, could you switch them over to that firmware?

Then you can use the same `  - platform: total_daily_energy` configuration on the Shelly side.
